ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/service/internal/RemoteFileServiceImpl.java

```diff
@@ -149,8 +154,16 @@ public void getFile(String fileId,
             endCoordinate = reqFile.getFileSize();
 
         // File Archive Type - Reject GPG
-        if (reqFile.getFileName().toLowerCase().endsWith("gpg")) {
-            throw new NotImplementedException(sessionId + "Please contact EGA to download this file.");
+        if (reqFile.getFileName().toLowerCase().endsWith("gpg") || (encryptionAlgorithm != null && !"aes256".equals(encryptionAlgorithm))) {
+            String noContentExceptionMessage = sessionId + "Please contact EGA to download this file.";
```

Comment:
The sessionId will appear concatenated with the text. You will also want to be more clear with information about the file that the user has tried to access.